pkg/services/object/put/distributed.go

```diff
@@ -331,7 +331,7 @@ func (x placementIterator) iterateNodesForObject(obj oid.ID, f func(nodeDesc) er
 					if e, _ := lastRespErr.Load().(error); e != nil {
 						err = fmt.Errorf("%w (last node error: %w)", err, e)
 					}
-					return errIncompletePut{singleErr: err}
+					wg.Wait()
```

Comment:
So we'll always return `nil` here? This doesn't seem to be correct.

Reply:
2 changes are mixed in one. Wait to fix data race - ok, but cotinuing the loop changes the behavior

Reply:
Looks like we can omit the `return` really, this will be handled. And not mess with `err` here as well.

Reply:
We can even not `wg.Wait()` then (it's done below anyway).

Reply:
> we can omit the return really

i need a final solution. you refused mine (or not?), then what should we do here?

> We can even not wg.Wait() then (it's done below anyway).

yes but then we immediately try the next node, which i probably useless since i expect pool overload with disk operations to be longer than the next `for` iteration